netbox/dcim/models/cables.py

```diff
@@ -529,46 +535,66 @@ def from_origin(cls, terminations):
                 object_to_path_node(t) for t in terminations
             ])
 
-            # Step 2: Determine the attached link (Cable or WirelessLink), if any
-            link = terminations[0].link
-            if link is None and len(path) == 1:
-                # If this is the start of the path and no link exists, return None
-                return None
-            elif link is None:
+            # Step 2: Determine the attached links (Cable or WirelessLink), if any
+            links = [termination.link for termination in terminations if termination.link is not None]
+            if len(links) == 0:
+                if len(path) == 1:
+                    # If this is the start of the path and no link exists, return None
+                    return None
                 # Otherwise, halt the trace if no link exists
                 break
-            assert type(link) in (Cable, WirelessLink)
+            assert all(type(link) in (Cable, WirelessLink) for link in links)
+
+            # Create set of links in path.  Cannot use list(set()) as it does this in a non-deterministic manner
```

Comment:
> Cannot use list(set()) as it does this in a non-deterministic manner

The implications of this comment aren't immediately apparent. Please extend it to elaborate on why the list of links must be ordered.

Separately, I'm not sure how it could be valid to ignore redundant links in the list if ordering is important. How do we handle a scenario where two pairs of ports are connected via cable A and another two pairs of ports are connected via cable B?

Reply:
You are absolutely corect here, I originally did it because of the way it was coded to not lose the cable position and diagram it as a simple fan-in/fan-out, however the cable svg logic should instead be modified to account for this.